### PR TITLE
MMU: Fix IsRAMAddress not working

### PR DESCRIPTION
**Pull request by sepalani**
This PR fixes the `IsRAMAddress` function returning true when the RAM is NULL. It can happen very rarely when the hardware is shutting down and there is something interacting with the RAM at the same time like some debugger widgets.

Ready to be reviewed.

### CODE REVIEW

**Comment by AdmiralCurtiss** [comment]
This smells race-y. Do you have an example of a problem caused by this?

**Comment by AdmiralCurtiss** [comment]
Aaah yeah okay that makes sense. And yeah, I agree, this patch looks sensible, I was just wondering if there wasn't a deeper issue hidden here.

File: Source/Core/Core/PowerPC/MMU.cpp [hunk 1]
```diff
@@ -665,14 +665,15 @@ static bool IsRAMAddress(u32 address, bool translate)
...
+  if (Memory::m_pRAM && segment == 0x0 && (address & 0x0FFFFFFF) < Memory::GetRamSizeReal())
     return true;
   else if (Memory::m_pEXRAM && segment == 0x1 &&
            (address & 0x0FFFFFFF) < Memory::GetExRamSizeReal())
     return true;
   else if (Memory::m_pFakeVMEM && ((address & 0xFE000000) == 0x7E000000))
     return true;
-  else if (segment == 0xE && (address < (0xE0000000 + Memory::GetL1CacheSize())))
+  else if (Memory::m_pL1Cache && segment == 0xE &&
+           (address < (0xE0000000 + Memory::GetL1CacheSize())))
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/MMU.cpp:676]: this should be braced given the condition travels more than one line

**Comment by AdmiralCurtiss** [comment]
You may want to add the same checks in `ReadFromHardware()` and `WriteToHardware()` too.

**Comment by sepalani** [comment]
@AdmiralCurtiss 
Wouldn't that add an additional overhead and defeat the purpose of the `IsRAMAddress` function?

**Comment by AdmiralCurtiss** [comment]
But in that case we're being quire inconsistent, ReadFromHardware() and WriteToHardware() do check `m_pEXRAM` and `m_pFakeVMEM`.

**Comment by sepalani** [comment]
You've a point, I'll add the check to them as well, then.

**Pull request merged**